### ISSUE TITLE
How to use babel-eslint with airbnb config?

### ISSUE THREAD

**Issue by screendriver** [babel/babel-eslint#298]
I want to use all ESLint rules from the [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript) in combination with your `babel-eslint` parser because the Airbnb guys does not support ES7+ features like `async / await`.

But I don't get it running. It complains about some `jsx-a11y` rules that were added in the latest Airbnb release:

```sh
3:1   error  Definition for rule 'jsx-a11y/img-uses-alt' was not found     jsx-a11y/img-uses-alt
3:1   error  Definition for rule 'jsx-a11y/redundant-alt' was not found    jsx-a11y/redundant-alt
3:1   error  Definition for rule 'jsx-a11y/valid-aria-role' was not found  jsx-a11y/valid-aria-
```

My `package.json` looks like

```js
...
  "devDependencies": {
    "babel-eslint": "^6.0.2",
    "eslint": "^2.8.0",
    "eslint-config-airbnb": "^7.0.0",
    "eslint-plugin-jsx-a11y": "^1.0.1",
    "eslint-plugin-react": "^5.0.1",
  }
...
```

any my `.eslintrc`

```json
{
  "parser": "babel-eslint",
  "extends": "airbnb",
  "rules": {
    "strict": 0
  }
}
```

If I remove `babel-eslint` parser from my `.eslintrc` it is running but complains about

```sh
55:8  error  Parsing error: Unexpected token async
```

(I am using `async` and `await` in my code and that's not support by Airbnb)

Any help is welcome :relaxed:

**Comment by fatfisz** [comment]
Just a shot in the dark, but shouldn't you also add this to the `.eslintrc`?
```
"plugins": [
  "jsx-a11y",
  "react"
], 
```

**Comment by screendriver** [comment]
Unfortunately no improvement

```js
{
  "parser": "babel-eslint",
  "extends": "airbnb",
  "plugins": [
    "jsx-a11y",
    "react"
  ],
  "rules": {
    "strict": 0
  }
}
```

shows me the same error.

**Comment by fatfisz** [comment]
Ok, I've installed the dependencies you have (except for "eslint-config-mip", which doesn't exist).
I didn't get any errors like yours after launching eslint.

Maybe what you need is to remove "node_modules" and fire `npm install` again?

Btw. I checked and the "plugins" section is already in the airbnb config, so it shouldn't change anything - my bad.

**Comment by screendriver** [comment]
I created a very simple [test project](https://github.com/screendriver/babel-eslint-error). Just clone it, run `npm install` and after that `./node_modules/.bin/eslint index.js` and you can see what I mean.

**Comment by fatfisz** [comment]
Ok, a mistake on my side - eslint ignored an empty file and that's why I didn't see any errors in my test project.

I've found an answer: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/CHANGELOG.md (note changes introduced in 1.0.0).

Basically, the airbnb config uses another version of `eslint-plugin-jsx-a11y`. You should watch out for warnings while installing, because you should have got these:
```
npm WARN eslint-config-airbnb@7.0.0 requires a peer of eslint-plugin-jsx-a11y@^0.6.2 but none was installed.
npm WARN eslint-config-airbnb@7.0.0 requires a peer of eslint-plugin-react@^4.3.0 but none was installed.
```

Just a thought: it would be better to create this issue in the airbnb config project, because warnings are concerned with it and not with `babel-eslint`.

Seems like airbnb people already have updated the dependencies and the rules, you just have to wait for the new version to be released (or use older versions of the plugins).

**Comment by screendriver** [comment]
Thank you for your help! 👍 

I understand. I will install the older dependencies and wait for a new Airbnb release.
